### ISSUE TITLE
[fix-bridge] OrderCancelReject instead of Reject?

### ISSUE THREAD

**Issue by mmitkevich** [mmitkevich/roq-rencap#37]
![image](https://user-images.githubusercontent.com/3052367/122794465-f7568e80-d2c4-11eb-9d62-e363c1f7880c.png)
```
I0621 11:57:05.620466 20031 session.cpp:1168] 8=FIX.4.4|9=0000278|35=W|49=roq-fix-bridge|56=tbricks-bitmex|34=5|52=20210621-15:57:05.620|262=e17e0d449c4dd2a|55=ETHUSD|207=bitmex|268=2|269=0|270=1921.04999999|271=3.00000000|272=20210621|273=15:53:58.477|336=bitmex|269=1|270=1980.95000000|271=5.00000000|272=20210621|273=15:53:58.477|336=bitmex|10=221|
I0621 11:57:05.620485 20031 reader.cpp:16] 8=FIX.4.4|9=129|35=e|49=tbricks-bitmex|56=roq-fix-bridge|34=6|52=20210621-15:57:05.619|324=e17e0d449c4dd34|55=XBTUSD|207=bitmex|263=1|336=bitmex|10=162|
I0621 11:57:05.620492 20031 session.cpp:1168] 8=FIX.4.4|9=0000130|35=f|49=roq-fix-bridge|56=tbricks-bitmex|34=6|52=20210621-15:57:05.620|324=e17e0d449c4dd34|55=XBTUSD|207=bitmex|336=bitmex|326=17|10=138|
I0621 11:57:05.620517 20031 reader.cpp:16] 8=FIX.4.4|9=209|35=V|49=tbricks-bitmex|56=roq-fix-bridge|34=7|52=20210621-15:57:05.619|262=e17e0d449c4dd3e|263=1|264=1|265=1|266=Y|267=10|269=0|269=1|269=3|269=4|269=5|269=6|269=7|269=8|269=9|269=2|146=1|55=XBTUSD|207=bitmex|10=185|
W0621 11:57:05.620539 20031 utils.h:72] Unable to map TRADING_SESSION_LOW_PRICE (fix::MDEntryType)
I0621 11:57:05.620541 20031 market.cpp:52] Market (id=182, exchange="bitmex", symbol="XBTUSD"): SUBSCRIBE session_id=3, md_req_id="e17e0d449c4dd3e", subsriptions=0x64000000000001ff
I0621 11:57:05.620551 20031 session.cpp:1168] 8=FIX.4.4|9=0000285|35=W|49=roq-fix-bridge|56=tbricks-bitmex|34=7|52=20210621-15:57:05.620|262=e17e0d449c4dd3e|55=XBTUSD|207=bitmex|268=2|269=0|270=32688.50000000|271=4000.00000000|272=20210621|273=15:56:51.172|336=bitmex|269=1|270=32689.00000000|271=300.00000000|272=20210621|273=15:56:51.172|336=bitmex|10=251|
I0621 11:57:08.109617 20031 session.cpp:1168] 8=FIX.4.4|9=0000311|35=X|49=roq-fix-bridge|56=tbricks-bitmex|34=8|52=20210621-15:57:08.109|262=e17e0d449c4dd2a|268=2|279=1|269=0|55=ETHUSD|207=bitmex|270=1921.04999999|271=3.00000000|272=20210621|273=15:57:07.909|336=bitmex|279=1|269=1|55=ETHUSD|207=bitmex|270=1982.34999999|271=1.00000000|272=20210621|273=15:57:07.909|336=bitmex|10=250|
I0621 11:57:09.109235 20031 session.cpp:1168] 8=FIX.4.4|9=0000311|35=X|49=roq-fix-bridge|56=tbricks-bitmex|34=9|52=20210621-15:57:09.109|262=e17e0d449c4dd2a|268=2|279=1|269=0|55=ETHUSD|207=bitmex|270=1921.04999999|271=3.00000000|272=20210621|273=15:57:08.248|336=bitmex|279=1|269=1|55=ETHUSD|207=bitmex|270=1982.45000000|271=5.00000000|272=20210621|273=15:57:08.248|336=bitmex|10=198|
I0621 11:57:33.031234 20031 reader.cpp:16] 8=FIX.4.4|9=241|35=D|49=tbricks-bitmex|56=roq-fix-bridge|34=8|52=20210621-15:57:33.030|11=A2uadofr3b3fm|453=1|448=RB331|447=D|452=3|1=RENCM_TCA|21=1|386=1|336=bitmex|55=XBTUSD|207=bitmex|54=1|60=20210621-15:57:33.029|38=7000|40=2|44=32689|59=0|58=000FNG751|10=002|
I0621 11:57:33.031444 20031 session.cpp:1168] 8=FIX.4.4|9=0000313|35=8|49=roq-fix-bridge|56=tbricks-bitmex|34=10|52=20210621-15:57:33.031|37=|11=A2uadofr3b3fm|17=........91053031428345|150=A|39=A|103=-1|1=A1|581=-1|55=XBTUSD|207=bitmex|54=1|40=2|38=7000.00000000|44=32689.00000000|59=0|32=0.00000000|336=bitmex|151=7000.00000000|14=0.00000000|6=0|60=19700101-00:00:00.000|851=-1|10=192|
I0621 11:57:33.105109 20031 session.cpp:1168] 8=FIX.4.4|9=0000253|35=9|49=roq-fix-bridge|56=tbricks-bitmex|34=11|52=20210621-15:57:33.105|37=bitmex-de3e2f95-a4ed-4826-a5c5-2a7e19f6a16a|11=A2uadofr3b3fm|41=|39=.|1=A1|434=.|58=Rejected: Account has insufficient Available Balance, 246422 XBt required\nSubmitted via API.|10=133|
I0621 11:57:33.105141 20031 session.cpp:1168] 8=FIX.4.4|9=0000462|35=8|49=roq-fix-bridge|56=tbricks-bitmex|34=12|52=20210621-15:57:33.105|37=bitmex-de3e2f95-a4ed-4826-a5c5-2a7e19f6a16a|11=A2uadofr3b3fm|17=........91053105118820|150=.|39=.|103=-1|1=A1|581=-1|55=XBTUSD|207=bitmex|54=1|40=2|38=7000.00000000|44=32689.00000000|59=0|32=0.00000000|336=bitmex|151=7000.00000000|14=0.00000000|6=0.00000000|60=20210621-15:57:33.072|58=Rejected: Account has insufficient Available Balance, 246422 XBt required\nSubmitted via API.|851=-1|10=176|
I0621 11:57:33.106944 20031 reader.cpp:16] 8=FIX.4.4|9=177|35=F|49=tbricks-bitmex|56=roq-fix-bridge|34=9|52=20210621-15:57:33.105|41=A2uaapx2upxvz|11=D2uadofr3b3fn|55=XBTUSD|207=bitmex|54=1|60=20210621-15:57:33.105|38=7000|58=000FNG751|10=192|
I0621 11:57:33.106957 20031 session.cpp:1168] 8=FIX.4.4|9=0000131|35=9|49=roq-fix-bridge|56=tbricks-bitmex|34=13|52=20210621-15:57:33.106|37=NONE|11=D2uadofr3b3fn|41=A2uaapx2upxvz|39=.|434=1|102=1|10=119|

```

**Comment by thraneh** [comment]
Was a bug.
I didn't branch on reject for NewOrderSingle vs OrderCancelRequest/OrderCancelReplaceRequest.
This is a special case under A.1.a (FIX 4.4 order state changes, shown in italics).
This is now fixed and a unit test has been added.

**Comment by mmitkevich** [comment]
Thanks, will retest
